templates/types/streaming/nextjs/app/components/ui/chat/index.ts

```diff
@@ -21,17 +21,18 @@ export type ImageData = {
 
 export type DocumentFileType = "csv" | "pdf" | "txt" | "docx";
 
-export type DocumentFileContent = {
-  type: "ref" | "text";
-  value: string[] | string;
+export type UploadedFileMeta = {
+  id: string;
+  name: string;
+  url?: string;
+  refs?: string[];
 };
 
 export type DocumentFile = {
-  id: string;
   filename: string;
   filesize: number;
   filetype: DocumentFileType;
-  content: DocumentFileContent;
+  metadata?: UploadedFileMeta;
```

Comment:
in which cases is metadata undefined? what happens then if we don't have an ID?

Reply:
It's undefined when it's not uploaded to backend, if the it don't have an id then the FE would throw error i guess.

Reply:
Do you remember that any use case that we have multiple upload files, i guess that use case will have error with the current id in metadata but As it looks to me that we only allow to upload one file at a time.